pkg/cmd/openshift-operators/apis/operators/v1alpha1/types.go

```diff
@@ -0,0 +1,86 @@
+	Enabled ManagementState = "Enabled"
+	// Unmanaged means that the operator is not taking any action related to the component
+	Unmanaged ManagementState = "Unmanaged"
+	// Disabled means that the operator is actively managing its resources and trying to remove all traces of the component
+	Disabled ManagementState = "Disabled"
+)
+
+type OperatorSpec struct {
+	// managementState indicates whether and how the operator should manage the component
+	ManagementState ManagementState `json:"managementState"`
+
+	// imagePullSpec is the image to use for the component.
+	ImagePullSpec string `json:"imagePullSpec"`
+
+	// version is the desired state in major.minor.micro-patch.  Usually patch is ignored.
+	Version string `json:"version"`
+
+	// logging contains glog parameters for the component pods.  It's always a command line arg for the moment
+	Logging LoggingConfig `json:"logging,omitempty"`
+}
+
+type LoggingConfig struct {
+	// level is passed to glog.
+	Level int64 `json:"level"`
+
+	// vmodule is passed to glog.
+	Vmodule string `json:"vmodule"`
+}
+
+type ConditionStatus string
+
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+
+	OperatorStatusTypeAvailable = "Available"
+	OperatorStatusTypeMigrating = "Migrating"
+	OperatorStatusTypeFailing   = "Failing"
+)
+
+type OperatorCondition struct {
+	Type               string          `json:"type"`
+	Status             ConditionStatus `json:"status"`
+	LastTransitionTime metav1.Time     `json:"lastTransitionTime,omitempty"`
+	Reason             string          `json:"reason,omitempty"`
+	Message            string          `json:"message,omitempty"`
+}
+
+// VersionAvailablity gives information about the synchronization and operational status of a particular version of the component
+type VersionAvailablity struct {
+	// version is the level this availbility applies to
+	Version string `json:"version"`
+	// updatedReplicas indicates how many replicas are at the desired state
+	UpdatedReplicas int32 `json:"updatedReplicas"`
+	// readyReplicas indicates how many replicas are ready and at the desired state
+	ReadyReplicas int32 `json:"readyReplicas"`
+	// errors indicates what failures are associated with the operator trying to manage this version
+	Errors []string `json:"errors"`
+}
+
+type OperatorStatus struct {
+	// conditions is a list of conditions and their status
+	Conditions []OperatorCondition `json:"conditions,omitempty"`
+
+	// state indicates what the operator has observed to be its current operational status.
+	State ManagementState `json:"state,omitempty"`
+	// task is a high level summary of what the controller is currently attempting to do.  It is high-level, human-readable
+	// and not guaranteed in any way. (I needed this for debugging and realized it made a great summary).
+	Task string `json:"task,omitempty"`
```

Comment:
Why not call it "Summary" or "Message" (I think it is called Message in Deployment.Status ;-))

Reply:
> Why not call it "Summary" or "Message" (I think it is called Message in Deployment.Status ;-))

This is a slightly different thing and may go away if we find a different place for debugging info.  Give it to me for now?

Reply:
i would just call it TaskSummary or something to emphasize that this is being used as text/message field, i'm fine with having this field for now.